### DISCUSSION TITLE
Scala 3

### DISCUSSION THREAD

**Discussion by MarekPikula** [SpinalHDL/SpinalHDL#799] [General]
What is the status of Scala 3 port? I can see that there is a branch, which is somewhat behind the dev branch.

**Comment by Dolu1990** [comment]
The status is, SpinalHDL core and the scala compiler plugin works, but it hit some Scala 3 compiler bugs in the lib XD
in short, in Scala 3 anonymous class are kind of not so good. I will check now how it goes with the very last version of scala 3

**Reply by Dolu1990** [reply]
Also, quite a few internal domain specific language syntax tricks aren't working anymore.

**Comment by Dolu1990** [comment]
For instance constructs as new Fragment(cloneOf(new Rawrrrr())) work well in scala 2, but in scala 3 => 

```

    class Rawrrrr() extends Bundle{
      val a,b,c = UInt(8 bits)
    }
    new Fragment(cloneOf(new Rawrrrr()))
    
/media/data/open/scala3/SpinalHDL/lib/src/main/scala/spinal/lib/Aaaaaaaaaaaaaaaaaaaaaa.scala:10:26
Found:    Rawrrrr
Required: spinal.core.Data & spinal.core.HardType[T]

where:    T is a type variable with constraint <: spinal.core.Data


The following import might make progress towards fixing the problem:

  import collection.Searching.search


    new Fragment(cloneOf(new Rawrrrr()))

```

**Reply by MarekPikula** [reply]
Oh, I see.

**Comment by goekce** [comment]
Here are newer discussions about this topic:

https://github.com/SpinalHDL/SpinalHDL/discussions/1087
https://github.com/chipsalliance/chisel/issues/2908
https://github.com/chipsalliance/chisel/issues/3967

Currently, Scala 3 support for SpinalHDL seems to be frozen. 

A central issue about this topic would be useful 🙂. @Dolu1990 if you want, I can open one.

**Reply by Dolu1990** [reply]
Hi,

Sure, fell free to open one. Then i will comment on it about the current progress status.

**Reply by goekce** [reply]
https://github.com/SpinalHDL/SpinalHDL/issues/1710
